src/main/java/org/mockito/internal/matchers/CapturingMatcher.java

```diff
@@ -9,13 +9,21 @@
 import static org.mockito.internal.exceptions.Reporter.noArgumentValueWasCaptured;
 
 import java.io.Serializable;
-import java.util.LinkedList;
+import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.List;
+import java.util.concurrent.locks.Lock;
+import java.util.concurrent.locks.ReadWriteLock;
+import java.util.concurrent.locks.ReentrantReadWriteLock;
 
 @SuppressWarnings("unchecked")
 public class CapturingMatcher<T> implements ArgumentMatcher<T>, CapturesArguments, VarargMatcher, Serializable {
 
-    private final LinkedList<Object> arguments = new LinkedList<Object>();
+    private final List<Object> arguments = new ArrayList<Object>();
```

Comment:
Good idea to use array list because the matcher offers API to access random elements by index, so your improvement makes the code faster :)